substrate/frame/revive/rpc/src/client.rs

```diff
@@ -321,14 +365,26 @@ impl Client {
 	) -> Result<(), ClientError> {
 		log::info!(target: LOG_TARGET, "🔌 Subscribing to new blocks ({subscription_type:?})");
 		self.subscribe_new_blocks(subscription_type, |block| async {
-			let (signed_txs, receipts): (Vec<_>, Vec<_>) =
-				self.receipt_provider.insert_block_receipts(&block).await?.into_iter().unzip();
+			// Special handling for genesis block (block 0)
+			// Genesis block is never stored on-chain, so we need to build it manually
+			let evm_block = if block.number() == 0 {
+				self.build_genesis_block(&block)
+					.await
+					.ok_or(ClientError::EthereumBlockNotFound)?
+			} else {
+				self.storage_api(block.hash()).get_ethereum_block().await?
+			};
```

Comment:
build_genesis_block, I guess this is mostly useful for test right where you might be starting a dev-node at block 0